doc/examples/big_data_ingestion.py

```diff
@@ -0,0 +1,146 @@
+# flake8: noqa: E501
+"""
+Large-scale shuffle ingestion pipeline
+===============================================
+
+In this example, you will learn how to build, deploy and scale up a machine
+learning shuffle ingestion pipeline using
+`Ray Dataset <https://docs.ray.io/en/latest/data/dataset.html>`_ and
+`Dataset Pipelines <https://docs.ray.io/en/latest/data/dataset-pipeline.html>`_.
+
+In particular, we will show you:
+
+* How to build a shuffle ingestion pipeline that loads, shuffles and feeds data
+  into distributed trainers in few lines of code;
+* How to scale the pipeline from ingesting 100MB data to
+  500GB data.
+
+.. image:: ../data/dataset-repeat-2.svg
+    :align: center
+
+"""
+
+###############################################################################
+# Python Setup
+# ------------
+#
+# First, we'll import all the libraries we'll be using. This step also helps us
+# verify that the environment is configured correctly. If any of the imports
+# are missing, an exception will be raised.
+
+import ray
+
+from typing import List
+from ray.data.dataset_pipeline import DatasetPipeline
+
+#######################################################################
+# Build shuffle ingestion pipeline
+# ----------------------------------
+#
+# A typical machine learning ingestion pipeline consists of the following 4
+# steps:
+#
+# 1. Load the training data from external storage;
+# 2. Iterate the data for multiple epochs;
+# 3. In each epoch, applying global shuffle to normalize the data;
+# 4. In each epoch, split the shuffled data into shards, and feed shards to
+#    distributed trainers;
+#
+# Let’s see how we implement such pipeline using Ray Dataset:
+
+
+def create_shuffle_pipeline(training_data_dir: str, num_epochs: int,
+                            num_shards: int) -> List[DatasetPipeline]:
+
+    return ray.data.read_parquet(training_data_dir)\
+        .repeat(num_epochs)\
+        .random_shuffle()\
+        .split(num_shards, equal=True)
```

Comment:
Nit: I think it's more readable this way
```suggestion
    return ray.data.read_parquet(training_data_dir) \
        .repeat(num_epochs) \
        .random_shuffle() \
        .split(num_shards, equal=True)
```